packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,336 @@
+    <nimble-table-column-boolean-text
+        field-name="testResult"
+        true-message="Pass"
+        false-message="Fail"
+    >
+        Test Result
+    <nimble-table-boolean-text>
+</nimble-table>
+```
+
+##### Example E: Enum Text Column
+
+We may try to provide an easy way for clients to map enum values to localized strings. Here is a concept for an API which probably has issues.
+
+```html
+<nimble-table>
+    <nimble-table-column-enum-text
+        field-name="status"
+    >
+        Status
+        <nimble-list-option slot="enum-string-0" value="0">Pass</nimble-list-option>
+        <nimble-list-option slot="enum-string-1" value="1">Fail</nimble-list-option>
+    <nimble-table-column-enum-text>
+</nimble-table>
+```
+
+**Pros:**
+
+-   Easy for clients to use since configuration is declarative. This means Blazor apps could do configuration from .NET code and not need to write JS.
+-   Consistent numeric formatting across apps
+-   Easy to enforce styling differences between string and numeric columns (e.g. right vs left text alignment)
+
+**Cons:**
+
+-   Requires Nimble team to design simple but powerful formatting and i18n APIs
+-   Can't solve some use cases like app-specific formatting logic
+
+**Implementation Cost:**
+
+-   API design and implementation for each new column type
+
+#### Alternative 4: Client provides custom column implementation for each use case
+
+Nimble already has a mechanism for clients to provide custom columns by deriving from a base class, specifying the data fields / template / styling, and registering the column type with Nimble. We could ask clients to use this mechanism for text column types.
+
+**Pros:**
+
+-   Zero implementation cost to Nimble team
+-   Powerful; clients can format data however they want, including via browser APIs which are i18n-friendly
+
+**Cons:**
+
+-   Higher burden on clients to specify template, styling, numeric formatting, etc in JS. This is especially burdensome in frameworks like Blazor.
+-   Difficult to enforce styling differences between string and numeric columns (e.g. right vs left text alignment)
+-   Potential cross-app inconsistency if formatting code isn't shared
+
+### Strawman Proposal
+
+For the sake of discussion my initial proposal is:
+
+1. We need to offer an approach for columns that require app-specific formatting logic to support above use cases like 5 (custom unit logic) and 8 (data of unknown type).
+    - I believe the cons of **Client provides a custom column implementation for each use case** are too great so we should invest in an approach that offers clients more consistency and simplicity.
+    - I'm leaning towards **Client specifies formatting function** over **Use `table-column-text`** because it more clearly encodes that the column is numeric, giving better type safety and allowing for more consistent styling. I'd like to do performance profiling to see how it impacts scroll performance before committing to this direction.
```

Comment:
Had discussed another option where the JS author could register new formatters and references them in column config. Something like:

The js implementation:
```js
await customElements.whenDefined('nimble-column-text-formatted');
customElements.get('nimble-column-text-formatted').registerCustomFormatter('my-formatter', text => `hello ${text}`);
```

The custom element usage:
```
<nimble-column-text-formatted custom-formatter="my-formatter"></nimble-column-text-formatted>
```

Pros:
- Alleviates my concern about exposing a property on the custom element definition that takes a JavaScript function as a parameter.
- Configuring the table stays declarative
- Should be easy to make that kind of formatter shareable

Cons: 
- No clear pattern for parameterizing the formatter compared to a custom element definition which can defines new attributes to configure formatting settings.
- Lack of parameterization encourages one-off application specific instances.

Reply:
I captured this as an alternative in the update in #1224, linking back to this comment for reference.